lib/src/auth/auth.dart

```diff
@@ -1,18 +1,99 @@
-/// ref https://developer.qiniu.com/kodo/manual/1644/security
+import 'dart:convert';
+import 'package:meta/meta.dart';
+import 'package:crypto/crypto.dart';
+import './put_policy.dart';
+
+class TokenInfo {
+  final String accessKey;
+  final PutPolicy putPolicy;
+  const TokenInfo(this.accessKey, this.putPolicy);
+}
+
+/// 提供用于鉴权的相关功能
+///
+/// 安全机制文档 ref https://developer.qiniu.com/kodo/manual/1644/security
 class Auth {
-  final String ak;
-  final String sk;
-  Auth(this.ak, this.sk);
+  /// 鉴权所需的 [AccessKey]
+  ///
+  /// 如何生成以及查看请参阅：
+  /// http://developer.qiniu.com/article/developer/security/index.html
+  /// 使用须知请查看：
+  /// https://developer.qiniu.com/kodo/kb/1334/the-access-key-secret-key-encryption-key-safe-use-instructions
+  final String accessKey;
+
+  /// 鉴权所需的 [SecretKey]
+  ///
+  /// 如何生成以及查看、使用等请参阅 [AccessKey] 的说明
+  final String secretKey;
 
-  String generateManageToken(dynamic params) {
-    throw UnimplementedError();
+  const Auth({
+    @required this.accessKey,
+    @required this.secretKey,
+  })  : assert(accessKey != null),
+        assert(secretKey != null);
+
+  /// 根据上传策略生成上传使用的 Token
+  ///
+  /// 具体的上传策略说明请参考 [PutPolicy] 模块
+  String generateUploadToken({
+    @required PutPolicy putPolicy,
+  }) {
+    assert(putPolicy != null);
+
+    var data = jsonEncode(putPolicy);
+    var encodedPutPolicy = base64Encode(utf8.encode(data));
+    var baseToken = generateManageToken(bytes: utf8.encode(encodedPutPolicy));
+    return '${baseToken}:${encodedPutPolicy}';
   }
 
-  String generateUploadToken(dynamic params) {
-    throw UnimplementedError();
+  /// 生成针对私有空间资源的下载 Token
+  ///
+  /// [key] 为对象的名称
+  /// [deadline] 有效时间，单位为秒，例如 1451491200
+  /// [bucketDomain] 空间绑定的域名，例如 http://test.bucket.com
+  String generateDownloadToken({
+    @required String key,
+    @required int deadline,
+    @required String bucketDomain,
+  }) {
+    assert(key != null);
+    assert(deadline != null);
+    assert(bucketDomain != null);
+
+    var downloadURL = '${bucketDomain}/${key}?e=${deadline}';
```

Comment:
约定一下，都用 final，类似与我们在 js 都用 const
```suggestion
    final downloadURL = '${bucketDomain}/${key}?e=${deadline}';
```

Reply:
我没有加上这条 lint 规则，你可以帮忙加上